solvers/fista_automatic_restart.py

```diff
@@ -0,0 +1,96 @@
+from benchopt import BaseSolver, safe_import_context
+
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from scipy import sparse
+
+
+class Solver(BaseSolver):
+    name = 'FISTA_automatic_restart'
+    stopping_strategy = "iteration"
```

Comment:
As the algorithm is implemented here, I would directly go for the callback, to avoid making it slower than it should.
The rational here is that you are given `cb` instead of `n_iter` and you should run `while cb(w):`.
See [here](https://benchopt.github.io/how.html?highlight=callback#python-solver) for an example with callback (second example).
```suggestion
    stopping_strategy = "callback"
```